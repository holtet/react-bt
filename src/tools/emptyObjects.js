export const malt = {
  comments: '',
  grams: 0,
  boughtDate: new Date(),
  maltDefinition: { name: '', description: '', manufacturer: '', country: '' },
};
export const maltType = {
  name: '',
  country: '',
  manufacturer: '',
  description: '',
};
export const yeast = {
  grams: 0,
  packages: 0,
  comments: '',
  boughtDate: new Date(),
  yeastType: { name: '', format: '' },
};
export const equipment = {
  boughtDate: new Date(),
  comments: '',
  amount: 0,
  equipmentType: {
    name: '',
    description: '',
  },
};
export const hop = {
  grams: 0,
  year: 2017,
  alpha: 0.0,
  comments: '',
  boughtDate: new Date(),
  hopType: { name: '', format: '' },
};
export const hopType = {
  name: '',
  manufacturer: '',
  country: '',
  description: '',
};
