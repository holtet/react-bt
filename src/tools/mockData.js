const yeastTypes = [
  {
    name: 'Saf5',
    manufacturer: 'Safale',
    country: 'England',
    format: 'DRY',
    _links: {
      self: {
        href: 'http://localhost:8080/yeastTypes/14',
      },
      yeastType: {
        href: 'http://localhost:8080/yeastTypes/14',
      },
    },
  },
  {
    name: 'WLP400',
    manufacturer: 'WyEast',
    country: 'Belgium',
    format: 'LIQUID',
    _links: {
      self: {
        href: 'http://localhost:8080/yeastTypes/15',
      },
      yeastType: {
        href: 'http://localhost:8080/yeastTypes/15',
      },
    },
  },
];
const hopTypes = [
  {
    name: 'Cascade',
    manufacturer: 'US Hop Co',
    country: 'USA',
    format: 'PELLETS',
    _links: {
      self: {
        href: 'http://localhost:8080/hopTypes/10',
      },
      hopType: {
        href: 'http://localhost:8080/hopTypes/10',
      },
    },
  },
  {
    name: 'hallertau',
    manufacturer: 'Hops GMBH',
    country: 'Germany',
    format: 'PELLETS',
    _links: {
      self: {
        href: 'http://localhost:8080/hopTypes/11',
      },
      hopType: {
        href: 'http://localhost:8080/hopTypes/11',
      },
    },
  },
];
const hops = [
  {
    year: 2017,
    alpha: 3.7,
    grams: 90,
    boughtDate: '2020-05-29',
    comments: '',
    openedDate: null,
    hopType: {
      name: 'Cascade',
      manufacturer: 'US Hop Co',
      country: 'USA',
      format: 'PELLETS',
    },
    type: {
      name: 'Cascade',
      manufacturer: 'US Hop Co',
      country: 'USA',
      format: 'PELLETS',
    },
    _links: {
      self: {
        href: 'http://localhost:8080/hops/12',
      },
      hop: {
        href: 'http://localhost:8080/hops/12{?projection}',
        templated: true,
      },
      type: {
        href: 'http://localhost:8080/hops/12/type',
      },
    },
  },
  {
    year: 2018,
    alpha: 5.3,
    grams: 15,
    boughtDate: '2020-05-29',
    comments: '',
    openedDate: null,
    hopType: {
      name: 'hallertau',
      manufacturer: 'Hops GMBH',
      country: 'Germany',
      format: 'PELLETS',
    },
    _links: {
      self: {
        href: 'http://localhost:8080/hops/13',
      },
      hop: {
        href: 'http://localhost:8080/hops/13{?projection}',
        templated: true,
      },
      type: {
        href: 'http://localhost:8080/hops/13/type',
      },
    },
  },
];
const apiHops = { _embedded: { hops: hops } };
const newHop = { id: null, comments: '' };
module.exports = {
  hopTypes,
  hops,
  apiHops,
  newHop,
  yeastTypes,
};
