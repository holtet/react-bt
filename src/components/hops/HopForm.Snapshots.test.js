import React from 'react';
import HopForm from './HopForm';
import renderer from 'react-test-renderer';
import { hops, hopTypes } from '../../tools/mockData';

it("sets submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <HopForm
      hop={hops[0]}
      hopTypes={hopTypes}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving
    />
  );
  //   console.log(tree.toJSON());
  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Save' when saving is false", () => {
  const tree = renderer.create(
    <HopForm
      hop={hops[0]}
      hopTypes={hopTypes}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving={false}
    />
  );
  //   console.log(tree.toJSON());
  expect(tree).toMatchSnapshot();
});
