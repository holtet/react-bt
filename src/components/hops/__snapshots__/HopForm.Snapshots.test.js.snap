// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' when saving is false 1`] = `
<div>
  <div
    className="overlay"
    style={
      Object {
        "width": 0,
      }
    }
  >
    <div
      className="sidenav-container"
    >
      <div
        className="text-center"
      >
        <h2>
          Hop
           changelog
        </h2>
      </div>
      <a
        className="closebtn"
        onClick={[Function]}
      >
        ×
      </a>
      <div
        className="list-group"
      >
        <table
          className="table table-dark"
          style={
            Object {
              "width": "95%",
            }
          }
        >
          <thead>
            <tr>
              <td>
                Time and date
              </td>
              <td>
                Field
              </td>
              <td>
                Changed from
              </td>
              <td>
                To
              </td>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <div
      className="card-header"
    >
      Hop Type
    </div>
    <div
      className="card-body"
    >
      <table
        className="table table-sm"
      >
        <tbody>
          <tr>
            <td>
              Name
            </td>
            <td>
              Cascade
            </td>
          </tr>
          <tr>
            <td>
              Country
            </td>
            <td>
              USA
            </td>
          </tr>
          <tr>
            <td>
              Format
            </td>
            <td>
              Pellets
            </td>
          </tr>
          <tr>
            <td>
              Manufacturer
            </td>
            <td>
              US Hop Co
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <form
    onSubmit={[MockFunction]}
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="grams"
      >
        grams
      </label>
      <div
        className="field"
      >
        <input
          className="form-control"
          name="grams"
          onChange={[MockFunction]}
          type="text"
          value="90"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="year"
      >
        year
      </label>
      <div
        className="field"
      >
        <input
          className="form-control"
          name="year"
          onChange={[MockFunction]}
          type="text"
          value="2017"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="alpha"
      >
        alpha
      </label>
      <div
        className="field"
      >
        <input
          className="form-control"
          name="alpha"
          onChange={[MockFunction]}
          type="text"
          value="3.7"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="comments"
      >
        comments
      </label>
      <div
        className="field"
      >
        <textarea
          className="form-control"
          name="comments"
          onChange={[MockFunction]}
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="boughtDate"
      >
        boughtDate
      </label>
      <div
        className="field"
      >
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className="form-control"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              readOnly={false}
              type="text"
              value="29.05.2020"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="openedDate"
      >
        openedDate
      </label>
      <div
        className="field"
      >
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className="form-control"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              readOnly={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="hopType"
      >
        hopType
      </label>
      <div
        className="field"
      >
        <select
          className="form-control"
          name="hopType"
          onChange={[MockFunction]}
        >
          <option
            value=""
          >
            Cascade | Pellets
          </option>
          <option
            value="http://localhost:8080/hopTypes/10"
          >
            Cascade | Pellets
          </option>
          <option
            value="http://localhost:8080/hopTypes/11"
          >
            hallertau | Pellets
          </option>
        </select>
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Save
    </button>
  </form>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Log
  </button>
</div>
`;

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<div>
  <div
    className="overlay"
    style={
      Object {
        "width": 0,
      }
    }
  >
    <div
      className="sidenav-container"
    >
      <div
        className="text-center"
      >
        <h2>
          Hop
           changelog
        </h2>
      </div>
      <a
        className="closebtn"
        onClick={[Function]}
      >
        ×
      </a>
      <div
        className="list-group"
      >
        <table
          className="table table-dark"
          style={
            Object {
              "width": "95%",
            }
          }
        >
          <thead>
            <tr>
              <td>
                Time and date
              </td>
              <td>
                Field
              </td>
              <td>
                Changed from
              </td>
              <td>
                To
              </td>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <div
      className="card-header"
    >
      Hop Type
    </div>
    <div
      className="card-body"
    >
      <table
        className="table table-sm"
      >
        <tbody>
          <tr>
            <td>
              Name
            </td>
            <td>
              Cascade
            </td>
          </tr>
          <tr>
            <td>
              Country
            </td>
            <td>
              USA
            </td>
          </tr>
          <tr>
            <td>
              Format
            </td>
            <td>
              Pellets
            </td>
          </tr>
          <tr>
            <td>
              Manufacturer
            </td>
            <td>
              US Hop Co
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <form
    onSubmit={[MockFunction]}
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="grams"
      >
        grams
      </label>
      <div
        className="field"
      >
        <input
          className="form-control"
          name="grams"
          onChange={[MockFunction]}
          type="text"
          value="90"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="year"
      >
        year
      </label>
      <div
        className="field"
      >
        <input
          className="form-control"
          name="year"
          onChange={[MockFunction]}
          type="text"
          value="2017"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="alpha"
      >
        alpha
      </label>
      <div
        className="field"
      >
        <input
          className="form-control"
          name="alpha"
          onChange={[MockFunction]}
          type="text"
          value="3.7"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="comments"
      >
        comments
      </label>
      <div
        className="field"
      >
        <textarea
          className="form-control"
          name="comments"
          onChange={[MockFunction]}
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="boughtDate"
      >
        boughtDate
      </label>
      <div
        className="field"
      >
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className="form-control"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              readOnly={false}
              type="text"
              value="29.05.2020"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="openedDate"
      >
        openedDate
      </label>
      <div
        className="field"
      >
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className="form-control"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              readOnly={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="hopType"
      >
        hopType
      </label>
      <div
        className="field"
      >
        <select
          className="form-control"
          name="hopType"
          onChange={[MockFunction]}
        >
          <option
            value=""
          >
            Cascade | Pellets
          </option>
          <option
            value="http://localhost:8080/hopTypes/10"
          >
            Cascade | Pellets
          </option>
          <option
            value="http://localhost:8080/hopTypes/11"
          >
            hallertau | Pellets
          </option>
        </select>
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={true}
      type="submit"
    >
      Saving...
    </button>
  </form>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Log
  </button>
</div>
`;
